### PR TITLE
Carthage compatibility for macOS, tvOS & watchOS

### PR DESCRIPTION
**Pull request by SimonRice**
This project as-is seems to be Carthage compatible for iOS.  This PR adds schemes & compatibility for macOS, tvOS & watchOS.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ script:
   - xcrun instruments -t 'Blank' -l 10 -w "iPhone 6s (10.0)"
   - swift build
   - swift test
-  - xcodebuild -scheme JSONUtilities -configuration Debug -sdk iphonesimulator10.0 -destination "OS=10.0,name=iPhone 6s" test -enableCodeCoverage YES
+  - xcodebuild -scheme "JSONUtilities iOS" -configuration Debug -sdk iphonesimulator10.0 -destination "OS=10.0,name=iPhone 6s" test -enableCodeCoverage YES
```
Comment (lucianomarisi) [inline .travis.yml]: I think that since we now have 4 schemes(and targets) we should probably build all 4 of them to make sure that they don't break

Comment (SimonRice) [reply .travis.yml]: Totally - I'm actually going to switch this to one scheme based on [the ideas from this article](http://promisekit.org/news/2016/08/Multiplatform-Single-Scheme-Xcode-Projects/) (& prompted by another project where I made similar scheme / project changes).  I'll certainly add tvOS as a test target & if your tests don't have UIKit, I'll add macOS too.  watchOS is another matter as it doesn't have XCTest, but I'll be happy to add a build check on here too.

**Comment by SimonRice** [comment]
~@lucianomarisi: feel free to review again - I've borrowed bits of `.travis.yml` from PromiseKit & you now have a single scheme for all platforms.~ Sorry, this is still on hold!

**Comment by lucianomarisi** [comment]
I'll hold then, let me know when it's ready for review.

**Comment by SimonRice** [comment]
Thank you @lucianomarisi - it looks like we're good to go once again.  You may wish to give running Carthage a spin when reviewing this PR & ensure all platforms have a framework build.

**Comment by lucianomarisi** [comment]
Gave Carthage a quick try and all of the targets work apart from the Mac one, do you have the same issue?

Looking into the Travis CI logs, I only see the tests being run for Mac and SPM. For example, https://travis-ci.org/lucianomarisi/JSONUtilities/jobs/226726271 does not look like it ran any tests, is this correct?

On another note, would it be worth updating the README to mention that Carthage support is now available for all 4 platforms?

**Comment by SimonRice** [comment]
I'll take a look at Carthage - I did occasionally have issues running `carthage install --platform ios,osx,tvos` with that specific combination, but then was fixed by running it for the individual platforms (or without specifying platform)!  I wonder if it's macOS causing the pain points for whatever reason, or if it's a Carthage bug - I'm happy to investigate this.

As for the tests, it looks to me like I somehow copy & hasted the commands for iOS & tvOS as these platforms aren't running the tests.  SPM & macOS look fine to me as the tests ran, as does watchOS as that platform doesn't do tests.  I'll certainly take a look & see how I can get the tests running again for iOS & tvOS.

At the very least, a Carthage badge would be well worth adding, along with your installation instructions.  I'll be happy to do this, or I can leave the honours down to you.

**Comment by lucianomarisi** [comment]
On the REAMDE side of things, I'd say you might just as well add the Carthage Badge and maybe update the installation section since this is related to this PR.

**Comment by SimonRice** [comment]
Hi @lucianomarisi - I've pushed a fix for Carthage on macOS & got the tests to run on Travis for iOS & tvOS.  

On top of that, I've slightly revamped the installation section.  You'll need to push a release tag to actually get the carthage bit working - feel free to change it to `github "lucianomarisi/JSONUtilities" "master"` if you don't plan on pushing a new release soon.

**Comment by lucianomarisi** [comment]
Awesome stuff, thank you!!

I'm looking to release V4 in the very near future, so that should work.

**Pull request merged**